hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -286,24 +286,24 @@ protected String getCommitActionType() {
 
   @Override
   protected void setCommitMetadata(HoodieWriteMetadata<HoodieData<WriteStatus>> result) {
-    result.setCommitMetadata(Option.of(CommitUtils.buildMetadata(result.getWriteStatuses().map(WriteStatus::getStat).collectAsList(),
+    List<HoodieWriteStat> writeStats = result.getWriteStatuses().map(WriteStatus::getStat).collectAsList();
+    result.setWriteStats(writeStats);
+    result.setCommitMetadata(Option.of(CommitUtils.buildMetadata(writeStats,
         result.getPartitionToReplaceFileIds(),
         extraMetadata, operationType, getSchemaToStoreInCommit(), getCommitActionType())));
   }
 
   @Override
   protected void commit(Option<Map<String, String>> extraMetadata, HoodieWriteMetadata<HoodieData<WriteStatus>> result) {
```

Comment:
@nsivabalan and @codope I noticed that we don't use `extraMetadata` at all in the method body. Is that intentional?

Reply:
yeah, I don't see a need for it. 
btw, in general, there are two code paths which we have been punting to refactor and fix it propery. 
auto cmmit enabled and auto commit disabled. 
this is auto commit enabled code path. 
both deltastreamer and spark datasource writer goes thru auto commit disabled code path. 
just fyi